FABulous/fabric_definition/Tile.py

```diff
@@ -309,3 +310,115 @@ def globalConfigBits(self) -> int:
+
+        Parameters
+        ----------
+        x_pitch : Decimal
+            Horizontal pitch between tracks (DBU)
+        y_pitch : Decimal
+            Vertical pitch between tracks (DBU)
+        x_pin_thickness_mult : Decimal
+            pin thickness multiplier in the horizontal direction
+        y_pin_thickness_mult : Decimal
+            pin thickness multiplier in the vertical direction
+        x_spacing : Decimal
+            pin spacing in the horizontal direction (DBU)
+        y_spacing : Decimal
+            pin spacing in the vertical direction (DBU)
+        frame_data_width : int, optional
+            frame data width in pins, by default 32
+        frame_stroble_width : int, optional
+            frame stroble width in pins, by default 20
+
+        Returns
+        -------
+        tuple[Decimal, Decimal]
+            (min_width, min_height) where:
+            - min_width: minimum width needed for north/south edge IO pins
+            - min_height: minimum height needed for west/east edge IO pins
+
+        Notes
+        -----
+        The minimum dimensions are calculated as:
+        - min_width = max(north_pins, south_pins) × x_pitch
+        - min_height = max(west_pins, east_pins) × y_pitch
+
+        These constraints prevent the LP solver from suggesting dimensions
+        that are physically impossible due to IO pin spacing requirements.
+        """
+        import itertools
+        from decimal import Decimal
+
+        # Count ports on each physical side
+        north_ports = len(
+            list(
+                itertools.chain.from_iterable(
+                    [
+                        list(itertools.chain.from_iterable(i.expandPortInfo("all")))
+                        for i in self.getNorthSidePorts()
+                    ]
+                )
+            )
+        )
+        south_ports = len(
+            list(
+                itertools.chain.from_iterable(
+                    [
+                        list(itertools.chain.from_iterable(i.expandPortInfo("all")))
+                        for i in self.getSouthSidePorts()
+                    ]
+                )
+            )
+        )
+        west_ports = len(
+            list(
+                itertools.chain.from_iterable(
+                    [
+                        list(itertools.chain.from_iterable(i.expandPortInfo("all")))
+                        for i in self.getWestSidePorts()
+                    ]
+                )
+            )
+        )
+        east_ports = len(
+            list(
+                itertools.chain.from_iterable(
+                    [
+                        list(itertools.chain.from_iterable(i.expandPortInfo("all")))
+                        for i in self.getEastSidePorts()
+                    ]
+                )
+            )
+        )
+
+        # Min width constrained by north/south edges
+        width_io_count = max(north_ports, south_ports) + frame_stroble_width
```

Comment:
```suggestion
        width_io_count = max(north_ports, south_ports) + frame_strobe_width
```
Please double check if I caught all occasions of this.